src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
-            case "Pickup": return State.Pickup;
-            case "PlacingA": return State.PlacingA;
-            case "PlacingB": return State.PlacingB;
-            case "PlacingC": return State.PlacingC;
-            case "PlacingD": return State.PlacingD;
-            case "PlacingE": return State.PlacingE;
+            case "Rest":
+                return State.Rest;
+            case "Pickup":
+                return State.Pickup;
+            case "PlacingA":
+                return State.PlacingA;
+            case "PlacingB":
+                return State.PlacingB;
+            case "PlacingC":
+                return State.PlacingC;
+            case "PlacingD":
+                return State.PlacingD;
+            case "PlacingE":
+                return State.PlacingE;
         }
 
         return ArmSubsystem.State.Rest;
     }
 
     public void periodic() {
-        matchActualState();
-        POS position;
-        PID shoulder_pid;
-        PID elbow_pid;
-
-        // If the state is not the same as the desired state, then run the periodic for that state.
-        if (m_state_actual != m_state_desired) {
-            shoulder_pid = SHOULDER_DEFAULT;
-            elbow_pid = ELBOW_DEFAULT;
-        } else {
-            shoulder_pid = SHOULDER_STEADY;
-            elbow_pid = ELBOW_STEADY;
-        }
-
-        switch (m_state_desired) {
-            case Pickup: position=PICKUP; break;
-            case PlacingA: position=PLACING_A; break;
-            case PlacingB: position=PLACING_B; break;
-            case PlacingC: position=PLACING_C; break;
-            case PlacingD: position=PLACING_D; break;
-            case PlacingE: position=PLACING_E; break;
-            default: position=REST;
-        }
-
-        m_elbow.getPIDController().setReference(position.elbow,
-                CANSparkMax.ControlType.kPosition,
-                elbow_pid.slot);
-
-        m_shoulder.getPIDController().setReference(position.shoulder,
-                CANSparkMax.ControlType.kPosition,
-                shoulder_pid.slot);
-
-        // Update the dashboard
-        SmartDashboard.putNumber("Shoulder Position", m_shoulder.getEncoder().getPosition());
-        SmartDashboard.putNumber("Elbow Position", m_elbow.getEncoder().getPosition());
-
-        handle.getEntry("/ArmSubsystem/DesiredState").setString(String.valueOf(m_state_desired));
-        handle.getEntry("/ArmSubsystem/ActualState").setString(String.valueOf(m_state_actual));
+        m_elbow.getPIDController().setReference(
+                ArmConstants.ELBOW_GEAR_RATIO(m_desiredState.elbow), ControlType.kPosition);
+        m_shoulder.getPIDController().setReference(
+                ArmConstants.SHOULDER_GEAR_RATIO(m_desiredState.shoulder), ControlType.kPosition);
     }
-
-    private static boolean inRange(double value, double target, double tolerance) {
-        return value >= target - tolerance && value <= target + tolerance;
-    }
-
-    private void matchActualState() {

```

Comment:
You can use your new functions to rewrite this but the functionality must remain the same.

Reply:
Why can't we use `atState`? This is not accurate based on my experience.

Reply:
A: What isn't accurate?
B: No we need to know when it's in transit so that while its transiting we don't change it which will put extensive strain on the arm and pose greater risk to incorrect operation.

Reply:
1. When I ran your code, the arm would go up to a point but stay in the transit state. It wouldn't transition to its actual state, which would break the workflow. I had to restart the robot code to reset it.
2. You will know if it's in transit because `atState` will be false.

Reply:
1. It shouldn't have been doing that. It was working exactly as intended when I left the code last. I can check again when the arm is functional.
2. That is fine with me. You can feel free to use the `atState` method if the outcome is the same as the intended behaviour.